compiler/pipes/preprocess-eq3.cpp

```diff
@@ -29,10 +29,11 @@ inline VertexPtr PreprocessEq3Pass::convert_eq3_null_to_isset(VertexPtr eq_op, V
   while (v->type() == op_index) {
     v = v.as<op_index>()->array();
   }
-  if (auto var_v = v.try_as<op_var>()) {
+  if (vk::any_of_equal(v->type(), op_var, op_instance_prop, op_array)) {
```

Comment:
Seems strange...
op_instance_prop seems strange, because it means "$a->x === null" here
op_array seems strange, because $a['x'] is not op_array — it's op_index (whereas op_array is just [1,2,3])

Reply:
I guess it's a lack of description, because 
op_instance_prop allows to replace both `$a->x === null`  and `$a->x["key"] === null`
op_array enables a case with a class constant, like `A::SOME_CONST_ARR["key"] === null` or even `["a" => "b"]["a"] === null`
so this preprocess works the same way for variables, instance pros and class constants.

Reply:
Nevertheless, we shouldn't replace "$a->x === null" with isset